pod-operation/src/components/wheel_encoder.rs

```diff
@@ -1,70 +1,94 @@
-use rppal::gpio::{Gpio, InputPin, Level};
-use std::time::Instant;
+use tokio::time::Instant;
+use rppal::gpio::{Gpio, InputPin};
 
-const PIN_ENCODER_A: u8 = 1;
-const PIN_ENCODER_B: u8 = 2;
 
+// GPIO pins for the encoder
+const PIN_ENCODER_A: u8 = 23; //need to move
+const PIN_ENCODER_B: u8 = 24; //need to move
+
+// Constants for calculations
+const DELTA_D: f32 = 1.0 / 16.0;
+
+type EncoderState = u8;
+type EncoderDiff = i8;
+
+// Function to encode the state
+// 00, 01, 10, 11
+fn encode_state(a: bool, b: bool) -> EncoderState {
+	((a as u8) << 1) + (a as u8 ^ b as u8)
+}
+
+// Function to calculate the difference in states
+// -1 => moving backwards
+// 0 => no movement
+// 1 => moving forward
+// 2 => Undersampling
+fn state_difference(p: EncoderState, q: EncoderState) -> EncoderDiff {
+	((p as i8 - q as i8 + 1) % 4 - 1) as EncoderDiff
+}
```

Comment:
Chore: similar to the other notes about casting, I think we could use `num_enum`'s traits to use From/Into.